docs/DeveloperGuide.md

```diff
@@ -343,77 +343,208 @@ b. Require developers to download those libraries manually (this creates extra w
+
+#### Use case: Mark a task/event as `done`
+
+**MSS**
+
+1. User requests to [display the active list](#list).
+2. User uses the index of the task/event of interest to request to mark it as `done`. Multiple indices can be entered to process more than one entries. 
+3. TaskCrusher moves the task/event from the active list to the expired list.
+
+Use case ends
 
 **Extensions**
 
-2a. The list is empty
+2a. The task/event is not found in the list
+> Use case ends
 
+2b. User decides not to mark a task/event.
 > Use case ends
 
-3a. The given index is invalid
+3a. The index is invalid
+> 3a1. TaskCrusher shows an error message <br>
+Use case resumes at step 2
+<br>
+#### Use case: manage overdue tasks and/or past events
 
-> 3a1. AddressBook shows an error message <br>
-  Use case resumes at step 2
+**MSS**
+
+1. User requests to [display the active list](#list) with the option to list only overdue tasks and/or past events relative to the current time.
+2. For each item in the displayed list, user uses the index to request to mark it as `done`, or dismiss it. Multiple indices may be entered to process more than one items. 
+3. If the user chose to dismiss an item, TaskCrusher prompts the user to enter the reason for it. 
+4. User enters the reason for dismissing a task/event. If user does not need to do so, user does nothing and proceeds.
+5. TaskCrusher moves the task/event from the active list to the expired list.
 
-{More to be added}
+Use case ends
+
+**Extensions**
+
+2a. There are no overdue tasks or past events
+> Use case ends
+
+2b. User quits the process
+> Use case ends
+
+3a. The index is invalid
+> Use case resumes at step 2
+
+4a. The reason is in the wrong format
+> Use case resumes at step 3
 
 ## Appendix C : Non Functional Requirements
 
-1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java `1.8.0_60` or higher installed.
-2. Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands)
-   should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should work on any desktop machine running on Windows 7 or later as long as it has Java `1.8.0_60` or higher installed.
+2. Must be fully functioning offline.
+3. No installation on the user machine should be required. 
+4. Should work stand-alone. 
+5. Must come with extensive JUnit error-testing.
+6. Must not rely on any relational databases.
+7. Must rely minimally on GUI. GUI is used only for output. 
+8. Must be able to support CLI.
+9. Supports one and only one user.  
+10. A user with an above-average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+11. Should be able to hold up to 1000 tasks without a noticeable decline in performance.
```

Comment:
what is "noticeable"? it may help to be specific like within 1 second, complete all actions etc.,